docs/guide/cookbook/module.md

```diff
@@ -419,30 +419,97 @@ Routes to be added must be _Array_ type even if it only has one element as you c
 
 ![route_liked_borderline](../images/route_liked.png)
 
-### 2-4. Recipe D (Use hooks)
-One of the most intuitive way to build a module is using hooks. Open source creators more than often intentionally leave hooks as many as possible to everywhere they think extendable for 3rd party developers to inject logic into the flow of program. Here we will look into how _Vue Storefront_ did it its way.
+### 2-4. Recipe D (Hooks inside a module)
+We have two main kinds of hooks `listener` and `mutator`. In this section, we will look into how to create hooks of both types.
 
+**How to create `hooks`?**
+
+1. Create the `hooks.ts` inside `./src/modules/example-module`.
+
+2. Import methods to create listener and mutator hooks from the `@vue-storefront/core/lib/hooks`:
+
+```js
+import { createListenerHook, createMutatorHook } from '@vue-storefront/core/lib/hooks'
+```
+
+3. Prepare our own `hooks`:
+```ts
+// abridged ...
+
+const {
+  hook: beforeExampleHook,
+  executor: beforeExampleExecutor
+} = createMutatorHook()
+
+// Also we can add some generic types to the hooks. The first argument is responsible for incoming data structure and the second one is type of returned data.
+
+const {
+  hook: beforeExampleHook,
+  executor: beforeExampleExecutor
+} = createMutatorHook<{ path: string, pathType: string, sizeX: number, sizeY: number }, { path: string }>()
+
+const {
+  hook: afterExampleHook,
+  executor: afterExampleExecutor
+} = createListenerHook()
+
+const exampleHooks = {
+  beforeExample: beforeExampleHook,
+  afterExample: afterExampleHook
+}
+const exampleHooksExecutors = {
+  beforeExample: beforeExampleExecutor,
+  afterExample: afterExampleExecutor
+}
+```
+
+4. Export created `hooks`:
+
+```js
+export {
+  exampleHooks,
+  exampleHooksExecutors
+}
+```
+
+**What is and why you need to call the `executor`?**
+
+In short `executor` is a function that will run all the collected hooks.
+Open source creators more than often intentionally leave hooks as many as possible to everywhere they think extendable for 3rd party developers to inject logic into the flow of the program.
+Ok, so why calling the `executor` is so important? It is important because without this all functions which you passed to hook will not be fired.
+
+**How to pass function to `hook`?**
+
+You just have to import the proper hook to any file inside your module and invoke it.
+
+```ts
+import { exampleHooks } from 'src/modules/example-module/hooks';
+
+exampleHooks.beforeExample(() => console.log('Do something!'))
```

Comment:
It should be mutator. Based on the doc it should return object containing `path: string`. Below you have a spread on that so the code will crash